rumqttd/src/router/routing.rs

```diff
@@ -1341,6 +1449,14 @@ fn forward_device_data(
     }
 
     outgoing.handle.try_send(()).ok();
+
+    // update the state of shared subscription
+    if let Some(share) = shared_group {
+        share.update_next_client();
```

Comment:
Thoughts on making this method take the cursor in? That way if it's ever used somewhere else we can make sure that cursor is always updated. Also looks a bit more nice

Reply:
`update_next_client()` is for updating the state of the _shared group_ right? What's the benefit of using the same method for updating cursor as well?. Plus there might be case where we just want to update the state but not the cursor.